test/terminal.js

```diff
@@ -0,0 +1,218 @@
+
+      terminal.mkdir('home');
+      terminal.ls();
+
+      expect(console.log.calledWith('home')).to.be.true;
+    });
+
+    it('Should return "home" as one of the directories in the root when "." parameter is given', function() {
+      const fileSystem = new FileSystem();
+      const terminal = new Terminal(fileSystem);
+
+      terminal.mkdir('home');
+      terminal.ls('.');
+
+      expect(console.log.calledWith('home')).to.be.true;
+    });
+
+    it('Should return "home" as one of the directories in the root when "/" parameter is given', function() {
+      const fileSystem = new FileSystem();
+      const terminal = new Terminal(fileSystem);
+
+      terminal.mkdir('home');
+      terminal.ls('/');
+
+      expect(console.log.calledWith('home')).to.be.true;
+    });
+
+    it('Should return "File or directory does not exist" when trying to list directories of an unexistent foo directory', function() {
+      const fileSystem = new FileSystem();
+      const terminal = new Terminal(fileSystem);
+
+      terminal.ls('foo');
+
+      expect(console.log.calledWith('File or directory does not exist')).to.be.true;
+    });
+
+    it('Should return all the directories and files when calling ls with the -r flag', function() {
+      const fileSystem = new FileSystem();
+      const terminal = new Terminal(fileSystem);
+
+      terminal.mkdir('foo');
+      terminal.mkdir('bar');
+      terminal.touch('root-1.txt');
+      terminal.touch('foo/foo-1.txt');
+      terminal.touch('foo/foo-2.txt');
+      terminal.touch('bar/bar-1.txt');
+      terminal.ls('-r');
+
+      expect(console.log.calledWith('/root')).to.be.true;
+      expect(console.log.calledWith('root-1.txt')).to.be.true;
+      expect(console.log.calledWith('foo')).to.be.true;
+      expect(console.log.calledWith('bar')).to.be.true;
+      expect(console.log.calledWith('/root/foo')).to.be.true;
+      expect(console.log.calledWith('foo/foo-1.txt'));
+      expect(console.log.calledWith('foo/foo-2.txt'));
+      expect(console.log.calledWith('/root/bar')).to.be.true;
+      expect(console.log.calledWith('bar/bar-1.txt'));
+    });
+
+  });
+
+  describe('Terminal cd: go to directory', function() {
+    it('Should return "/root/home" as the new current directory after going to "home" directory', function() {
+      const fileSystem = new FileSystem();
+      const terminal = new Terminal(fileSystem);
+
+      terminal.mkdir('home');
+      terminal.cd('home');
+      terminal.pwd();
+
+      expect(console.log.calledWith('/root/home')).to.be.true;
+    });
+
+    it('Should return "/root" as the current directory when calling cd with "." parameter', function() {
+      const fileSystem = new FileSystem();
+      const terminal = new Terminal(fileSystem);
+
+      terminal.cd('.');
+      terminal.pwd();
+
+      expect(console.log.calledWith('/root')).to.be.true;
+    });
+
+    it('Should return "guille" as one of the directories when go to "home" directory', function() {
```

Comment:
Add a description of what you are trying to test beyond what exactly the test does. Imagine that instead of creating "guille", you create "pepe" you should change the description of the test (not the behaviour you want in your code). 
Something like: 
`Show folders from a newly created folder`